dev/test/query.ts

```diff
@@ -3677,4 +3747,94 @@ describe('query resumption', () => {
     const actualDocumentIds = snapshots.map(snapshot => snapshot.id);
     expect(actualDocumentIds).to.eql(documentIds);
   });
+
+  it('resuming queries with a cursor should respect the original query limit', async () => {
```

Comment:
These two additional tests look fantastic. My only suggestion would be to add a 3rd request, to make sure that the final result is still composed correctly even if it took 3 individual requests to the backend.

Reply:
Good idea. Done.